core/Cargo.toml

```diff
@@ -28,6 +28,12 @@ ark-bn254 = { workspace = true, features = ["curve"], optional = true }
 ark-relations = { workspace = true, optional = true }
 ark-serialize = { workspace = true, optional = true }
 
+# serde support dependencies
+serde = { workspace = true, features = ["derive"], optional = true }
+serde_json = { workspace = true, optional = true }
```

Comment:
`serde_json` is a dev-dep.
The deny(unused_crate_dependencies) lint should catch this, but only if you don't use tricks like `use ... as _;'